src/explorer/SiteTreeItem.ts

```diff
@@ -166,6 +157,28 @@ export abstract class SiteTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
         }
     }
 
+    public async enableScmDoBuildDuringDeploy(fsPath: string, runtime: string): Promise<void> {
+        const zipIgnoreFolders: string[] = this.getIgnoredFoldersForDeployment(runtime);
+        let oldSettings: string[] | string | undefined = workspace.getConfiguration(constants.extensionPrefix, Uri.file(fsPath)).get(constants.configurationSettings.zipIgnorePattern);
+        if (!oldSettings) {
+            oldSettings = [];
+        } else if (typeof oldSettings === "string") {
+            oldSettings = [oldSettings];
+            // settings have to be an array to concat the proper zipIgnoreFolders
+        }
+        const newSettings: string[] = oldSettings;
+        for (const folder of zipIgnoreFolders) {
```

Comment:
Yeah I don't like the check for 'pathExists'. There's no harm in ignoring a folder that doesn't exist